packages/core/tests/controllers/EnsController.test.ts

```diff
@@ -0,0 +1,45 @@
+import { describe, expect, it, vi } from 'vitest'
+import { EnsController } from '../../index.js'
+
+// -- Setup --------------------------------------------------------------------
+const TEST_NAME = {
+  name: 'test',
+  registered: true,
+  updated: 1,
+  addresses: [
+    {
+      '0x123': {
+        address: '0x123',
+        created: '1'
+      }
+    }
+  ],
+  attributes: []
+}
+vi.mock('../../src/controllers/BlockchainApiController.js', async importOriginal => {
+  const mod =
+    await importOriginal<typeof import('../../src/controllers/BlockchainApiController.js')>()
+
+  return {
+    ...mod,
+    BlockchainApiController: {
+      lookupEnsName: async (name: string) => {
+        if (name !== 'test') {
+          throw new Error('Error resolving ENS name')
+        }
+
+        return Promise.resolve(TEST_NAME)
+      }
+    }
+  }
+})
+
+// -- Tests --------------------------------------------------------------------
+describe('EnsController', () => {
+  it('should have valid default state', () => {
```

Comment:
Seems we don't have many tests covered. We have functions like  `validateName`,  `parseEnsApiError` that could be unit tested